src/python/pants/help/help_printer.py

```diff
@@ -33,7 +27,7 @@ def __init__(
         self,
```

Comment:
The `HelpPrinter` class should probably be renamed (and maybe moved out of the `help` package) as well? Maybe to something like `NativeGoalHandler` or something?

Reply:
Moved to `/option/`? Or somewhere else entirely

Reply:
What's being introduced here is a general way to add "native" or "builtin" Goals.

They have that extra word (TBD) in their name order to differentiate them from [`@goal_rule`s](https://www.pantsbuild.org/docs/rules-api-goal-rules), which are how plugins add non-builtin Goals.

Given that, a few good top-level packages might be `pants.goal` or `pants.bin`...?

Reply:
Oh, ok -- goal makes the most sense to me, then.
I'll move the NativeGoalRequest class there as well.

Reply:
Given how much of the `HelpPrinter` is just printing help messages, could it make more sense to create a `NativeGoalHandler` that `HelpPrinter` inherits from, but ultimately leaving `HelpPrinter` in the help package?

Reply:
Yea, possibly. But in general: [prefer composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance). So if you were going to have separate classes for `NativeGoalHandler` and `HelpPrinter`, I'd suggest not having them inherit from one another, and instead have `NativeGoalHandler` just delegate relevant goals to the `HelpPrinter`.

Reply:
I think we want to keep `HelpPrinter` and do _not_ rename it, but instead add `BuiltinGoalPrinter` to `goal/builtin_goal.py` with generic printing functionality like knowing whether to use colors and the terminal width. This would then be used by `HelpPrinter`. 

You would be identifying in the current `HelpPrinter` what code is generic to printing to the console, and what is specific to the `help` system. Create a new class `BuiltinGoalPrinter` with only the generic stuff. The constructor for `HelpPrinter` then takes an instance of that, and delegates to it (composition over inheritance).

@wilsonliam does that suggestion make sense?

Reply:
It does; though I think at the moment (from memory of the help message PRs back in like February) most of the code is pretty specific to help messages. I'll revisit and refactor some

Reply:
Cool, it's possible you will need to define new methods on `BuiltinGoalPrinter` that you did not have on `HelpPrinter` currently. FYI see the somewhat related `Console` type:

https://github.com/pantsbuild/pants/blob/b73fe2816c62d265975122234013c08df9f9f4dd/src/python/pants/engine/console.py#L15-L101

Reply:
It sounds like there is a bit more going to happen around `BuiltinGoalPrinter` before landing this? Or were you thinking of that for followups?

Also, naming nit: not all `BuiltinGoal`s will be about Printing things: the (eventual) garbage collection that is being added in #11167 will be intended to cause a side-effect (garbage collecting things).

Reply:
Correct -- working on that atm; should be good by EOD or sometime tomorrow but wanted to push the complete stuff prior.

Given that the garbage collection component would probably still involve printing things (or future `BuiltinGoal`s), I think it makes sense to have those utils separated & reimplementing `HelpPrinter` to use that model. The `HelpPrinter` is a bit dense so separating is taking a little longer than I would've liked. That'll be the last component of this PR though.